src/content.ts

```diff
@@ -2484,6 +2460,82 @@ function addPageListeners(): void {
     document.addEventListener("yt-navigate-start", resetValues);
     document.addEventListener("yt-navigate-finish", refreshListners);
     window.addEventListener("message", windowListenerHandler);
+    window.addEventListener("load", insertSBIconDefinition)
+}
+
+function setUpThumbnailObserver() {
+    if (thumbnailObserver) {
+        thumbnailObserver.disconnect();
+        thumbnailObserver = null;
+    }
+    if (Config.config.fullVideoSegments) {
+        thumbnailObserver = new MutationObserver(thumbnailObserverHandler);
+        thumbnailObserver.observe(document, {childList: true, subtree: true})
```

Comment:
This is going to be called for _every_ child list change. I wonder if there's a better option here

Reply:
Probably the best option would be to somehow hook into the thumbnail custom element...